File: src/main/java/virtualMachine/vm_instruction_processing/InstructionProcessor.java
package virtualMachine.vm_instruction_processing;

import virtualMachine.stack.memory.VirtualMemory;
import virtualMachine.stack.types.instruction.Instruction;

public interface InstructionProcessor {
    /**
     * processes a command and interacts with memory
     * @param instruction e.g. push constant 1234
     * @param virtualMemory stores a functions local variables, stack and global 'heap' memory
     */
    void processInstruction(Instruction instruction, VirtualMemory virtualMemory);
}

File: src/main/kotlin/virtualMachine/stack/vm_instruction_parsing/vm_instruction_processing/InstructionProcessor.kt
package virtualMachine.stack.vm_instruction_parsing.vm_instruction_processing

import virtualMachine.stack.memory.VirtualMemory
import virtualMachine.stack.vm_instruction_parsing.InstructionStack

interface InstructionProcessor {
    fun processInstruction(instruction: String,
                           instructionStack: InstructionStack,
                           virtualMemory: VirtualMemory)
}
